go/mysql/datetime/datetime.go

```diff
@@ -580,29 +584,25 @@ func (dt *DateTime) addInterval(itv *Interval) bool {
 			return false
 		}
 
-		nsec := dt.Time.Nanosecond() + itv.nsec
-		sec := dt.toSeconds() + itv.toSeconds() + (nsec / int(time.Second))
-		nsec = nsec % int(time.Second)
-
-		if nsec < 0 {
-			nsec += int(time.Second)
-			sec--
-		}
+		dur := dt.toDuration()
+		dur += itv.toDuration()
+		days := time.Duration(0)
+		if !dt.Date.IsZero() {
```

Comment:
Rounding shouldn't move days if we're a Time instance. We then can have more than 24 hours.